src/main/java/org/terracotta/statistics/GeneralOperationStatistic.java

```diff
@@ -53,26 +52,44 @@
    */
   @Override
   public long count(T type) {
-    return counts.get(type).sum();
+    return counts[type.ordinal()].sum();
   }
 
   @Override
   public long sum(Set<T> types) {
     long sum = 0;
     for (T t : types) {
-      sum += counts.get(t).sum();
+      sum += counts[t.ordinal()].sum();
     }
     return sum;
   }
 
   @Override
   public void end(T result) {
-    counts.get(result).increment();
+    counts[result.ordinal()].increment();
     super.end(result);
   }
 
   @Override
   public String toString() {
-    return counts.toString();
```

Comment:
```
return stream(counts).map(constants[i] + "=" + counts[i]).collect(joining(", ", "[", "]"));
```

Reply:
Almost working ;-) But you are right. I've changed it.